src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,142 @@
+import { $$, ComponentOptions, IComponentBindings, Initialization, IQueryResult, IResultListOptions, IFieldOption } from 'coveo-search-ui';
+
+/**
+ * Interface for the object data returned from remote action.
+ *
+ * @export
+ * @interface IObjectData
+ */
+export interface IObjectData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    objectData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the remote action method.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs {
+    data: IObjectData;
+}
```

Comment:
I think the name of those interfaces is a bit unprecise, I would maybe prefix them with 'ResultListAugmenter' (augmenter as https://en.wiktionary.org/wiki/augmenter not, the french word for 'increase').

I'm not sure about ResultListAugmenter but, something of the like prolly.